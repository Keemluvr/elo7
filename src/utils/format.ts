/**
 * Capitalizes the first letter of the input string and converts the rest to lowercase.
 *
 * @param {string} string - The input string to be capitalized.
 * @return {string} The capitalized string.
 */
export const capitalize = (string: string) =>
  string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();

/**
 * Capitalizes each word in a string, including words separated by hyphens.
 *
 * @param {string} string - The input string to be capitalized.
 * @return {string} The capitalized string with each word capitalized and words separated by hyphens preserved.
 */
export const capitalizeWords = (string: string) => {
  return string
    .split(" ")
    .map((word) => {
      if (word.includes("-")) {
        const parts = word.split("-");
        return parts.map((part) => capitalize(part)).join("-");
      }
      return capitalize(word);
    })
    .join(" ");
};

/**
 * Formats a location string into a city, state, and country.
 *
 * @param {string} location - The location string to format.
 * @return {string} The formatted location string.
 */
export const formatLocation = (location: string) => {
  const [city, state, country] = location.split(", ");
  if (!city || !country) return "";
  return `${capitalizeWords(city)}, ${capitalizeWords(country)}`;
};

/**
 * Replaces specific substrings in a given string with their corresponding uppercase versions.
 *
 * @param {string} string - The input string to be processed.
 * @return {string} The modified string with the special cases replaced.
 */
export const specialCases = (string: string) => {
  return string
    .replace(/fp&a/gi, "FP&A")
    .replace(/Android/gi, "Android")
    .replace(/Ios/gi, "iOS")
    .replace(/devops/gi, "DevOps")
    .replace(/Sre/gi, "SRE");
};
