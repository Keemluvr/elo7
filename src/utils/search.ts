interface Data<T> {
  [key: string]: T[];
}

/**
 * Searches the data for entries that match all the terms in the query.
 *
 * @param {Data<T>} data - The data to search through.
 * @param {string | null} query - The search query.
 * @param {(keyof T)[]} properties - The properties to search within.
 * @return {Data<T> | undefined} The filtered data based on the search query.
 */
export const searchByCombinedTerms = <T>(
  data: Data<T>,
  query: string | null,
  properties: (keyof T)[]
): Data<T> | undefined => {
  if (!query) return;

  const terms = query.toLowerCase().split(" ");
  const filteredData: Data<T> = {};

  for (const key in data) {
    const filteredEntries = data[key].filter((entry: T) => {
      const entryString = properties
        .map((prop) => (entry[prop] as string).toLowerCase())
        .join(" ");
      return terms.every((term) => entryString.includes(term));
    });

    if (filteredEntries.length > 0) {
      filteredData[key] = filteredEntries;
    }
  }
  return Object.keys(filteredData).length > 0 ? filteredData : undefined;
};
