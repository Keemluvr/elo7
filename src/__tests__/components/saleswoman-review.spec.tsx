import { render, screen } from "@testing-library/react";
import { verifyAttributes } from "../utils/verifyAttributes";
import SaleswomanReview from "@/components/saleswoman-review";
import "@testing-library/jest-dom";

let container = {} as DocumentFragment;

describe("SaleswomanReview component", () => {
  beforeEach(() => {
    const { asFragment } = render(<SaleswomanReview />);
    container = asFragment();
  });

  describe("Section", () => {
    test("should render the section component", () => {
      const sectionElement = screen.getByRole("region", {
        name: "seção da palavra da vendedora",
      });
      expect(sectionElement).toBeInTheDocument();
    });

    test("should render the section with correct tag name", () => {
      const sectionElement = screen.getByRole("region", {
        name: "seção da palavra da vendedora",
      });
      expect(sectionElement.tagName).toBe("SECTION");
    });
  });

  describe("Image", () => {
    test("should render the image", () => {
      const imageElement = screen.getByRole("img", {
        name: "imagem da vendedora",
      });
      expect(imageElement).toBeInTheDocument();
    });

    test("should render the image with correct properties", () => {
      const imageElement = screen.getByRole("img", {
        name: "imagem da vendedora",
      });
      const expectedSrc =
        "/_next/image?url=%2Fimages%2Fsaleswoman.webp&w=3840&q=100";
      const attributes = [
        ["src", expectedSrc],
        ["loading", "lazy"],
        [
          "alt",
          "Uma mulher jovem trabalha em uma oficina de artesanato, concentrada em suas atividades. A mesa está repleta de ferramentas e materiais de trabalho, destacando um ambiente criativo e artesanal.",
        ],
      ];
      verifyAttributes(imageElement, attributes);
      expect(imageElement).toHaveClass("saleswoman-review-image");
      expect(imageElement).toHaveAttribute("srcset");
      expect(imageElement.getAttribute("srcset")).not.toBe("");
    });
  });

  describe("Description", () => {
    describe("Title", () => {
      test("should render the title", () => {
        const titleElement = screen.getByText("Palavra da vendedora");
        expect(titleElement).toBeInTheDocument();
      });

      test("should render the title with  with correct tag name", () => {
        const titleElement = screen.getByText("Palavra da vendedora");
        expect(titleElement.tagName).toBe("H2");
      });

      test("should render the title with correct properties", () => {
        const titleElement = screen.getByRole("heading", {
          name: "Palavra da vendedora",
        });
        expect(titleElement).toHaveClass("saleswoman-review-title");
      });
    });

    describe("Subtitle", () => {
      test("should render the subtitle", () => {
        const subtitleElement = screen.getByText("Sed rutrum condimentum");
        expect(subtitleElement).toBeInTheDocument();
      });

      test("should render the subtitle with correct properties", () => {
        const subtitleElement = screen.getByText("Sed rutrum condimentum");
        expect(subtitleElement).toHaveClass("saleswoman-review-subtitle");
      });
    });
  });

  test("matches snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
