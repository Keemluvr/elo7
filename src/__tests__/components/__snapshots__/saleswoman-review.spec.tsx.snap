// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleswomanReview component matches snapshot 1`] = `
<div>
  <section
    aria-label="seção da palavra da vendedora"
    class="grid-container saleswoman-review-wrapper"
  >
    <div
      class="grid-item-full-width saleswoman-review-content"
    >
      <div
        class="saleswoman-review-image-wrapper"
      >
        <img
          alt="Uma mulher jovem trabalha em uma oficina de artesanato, concentrada em suas atividades. A mesa está repleta de ferramentas e materiais de trabalho, destacando um ambiente criativo e artesanal."
          aria-label="imagem da vendedora"
          class="saleswoman-review-image"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fsaleswoman.webp&w=3840&q=100"
          srcset="/_next/image?url=%2Fimages%2Fsaleswoman.webp&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fsaleswoman.webp&w=3840&q=100 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="saleswoman-review-description"
      >
        <hgroup
          aria-label="cabecalho da avaliacao da vendedora"
        >
          <h2
            class="saleswoman-review-title"
          >
            Palavra da vendedora
          </h2>
          <p
            class="saleswoman-review-subtitle"
          >
            Sed rutrum condimentum
          </p>
        </hgroup>
        <p>
          Donec in vestibulum elit. Aliquam finibus facilisis elit, sit amet malesuada nibh tempor sed. Aliquam consequat ultrices fringilla. Sed id quam sollicitudin mi ultricies feugiat a vel velit. Pellentesque finibus vel tortor sed hendrerit. Vestibulum eu sem sapien. Nullam mollis, leo ut finibus euismod, arcu eros aliquam augue, in congue neque nulla vehicula purus.
        </p>
      </div>
    </div>
  </section>
</div>
`;
